### PR TITLE
[Backport]Testcase for detecting duplicates in Probe

### PR DESCRIPTION
**Pull request by shirishaganta**
Added test case to detect duplicates in probe

Signed-off-by: Shirisha Ganta <shiganta@in.ibm.com>

### CODE REVIEW

**Comment by shirishaganta** [comment]
JOB ID     : 29ae5b80ac959fb8d60c7d9f65659f419622b6b0
JOB LOG    : /home/avocado-fvt-wrapper-py2/results/job-2019-11-19T11.10-29ae5b8/job.log
 (1/1) perf_probe.py:PerfProbe.test_probe: FAIL: perf probe is placing multiple probe at the same location  (4.11 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 4.43 s
JOB HTML   : /home/avocado-fvt-wrapper-py2/results/job-2019-11-19T11.10-29ae5b8/results.html

**Comment by PraveenPenguin** [comment]
@nasastry one help are we need this against py2-lts as thinking new test case should only part of master python3 complaint only bug fix can be done against py2-lts 
@narasimhan-v thought please

**Comment by narasimhan-v** [comment]
@PraveenPenguin I would decide that on a per-test basis, as there might be a requirement to have the coverage in older distros as well, but that should be very rare.

**Comment by PraveenPenguin** [comment]
> @PraveenPenguin I would decide that on a per-test basis, as there might be a requirement to have the coverage in older distros as well, but that should be very rare.

yes my point is we should not encourgae to send new Test case against py2-lts but we can decide on case by case

**Comment by narasimhan-v** [comment]
@PraveenPenguin Right. Probably, a method we can follow is: raising an issue for adding a new test to py2-lts branch where we can all discuss. Once it is decided to add the test to py2-lts, the pull request can then be sent.

**Comment by nasastry** [comment]
@PraveenPenguin @narasimhan-v 
There is a bug in all distributions and wanted to automate that to check whether that bug exist or not. This test case applies to older distributions under Python2 as well. In this case, request you to add this test case to py-lts branch and further requests we can discuss via raising an issue. Thanks!!

**Comment by PraveenPenguin** [comment]
@nasastry sure thanks for helping, we can make this in and later we can go with the issue as @narasimhan-v pointed as you know LTS  only meant for the only bug fixes

File: perf/perf_duplicate_probe.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                      by %s" % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def test_probe(self):
+        output = process.run("perf probe select_task_rq_fair:44", sudo=True)
+        if 'select_task_rq_fair' in output.stderr and 'select_task_rq_fair_' in output.stderr:
+             fail_flag = 1
+        output = process.run("cat /sys/kernel/debug/tracing/kprobe_events", sudo=True)
```
Comment (PraveenPenguin) [inline perf/perf_duplicate_probe.py]: please use genio lib for file access

File: perf/perf_duplicate_probe.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+class PerfProbe(Test):
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf and systemtap-sdt-devel
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
```
Comment (PraveenPenguin) [inline perf/perf_duplicate_probe.py]: please use this as local scope variable as it not used any where aprt of setup sub routine

Comment (PraveenPenguin) [reply perf/perf_duplicate_probe.py]: and try to eliminate more variable defining as it increase grabage collector job

Comment (shirishaganta) [reply perf/perf_duplicate_probe.py]: sure Praveen

**Comment by shirishaganta** [comment]
SLES:
JOB ID     : ec33f88b8936e04535d2dc6ccfbfbf41356cf862
JOB LOG    : /home/avocado-fvt-wrapper-py2/results/job-2019-11-21T12.09-ec33f88/job.log
 (1/1) perf_duplicate_probe.py:PerfProbe.test_probe: FAIL: perf probe is placing multiple probe at the same location  (1.01 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1.57 s
JOB HTML   : /home/avocado-fvt-wrapper-py2/results/job-2019-11-21T12.09-ec33f88/results.html

RHEL:
JOB ID     : adced3ab783517a0f5755084175be6ce5b34485e
JOB LOG    : /home/avocado-fvt-wrapper-py2/results/job-2019-11-21T09.40-adced3a/job.log
 (1/1) perf_duplicate_probe.py:PerfProbe.test_probe: FAIL: perf probe is placing multiple probe at the same location  (3.44 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 3.82 s
JOB HTML   : /home/avocado-fvt-wrapper-py2/results/job-2019-11-21T09.40-adced3a/results.html

**Comment by PraveenPenguin** [comment]
merging as it already in master

**Comment by PraveenPenguin** [comment]
@nasastry @narasimhan-v thanks for input in this PR , @shirishaganta thanks for PR

**Pull request merged**